frontend/src/pages/ProtectedRoute.jsx

```diff
@@ -0,0 +1,49 @@
+import React, { useState, useEffect } from "react";
+import axios from "axios";
+import { Navigate} from "react-router-dom";
+
+const VALIDATE_STATE = {
+    IDLE: 'idle',
+    VALIDATED: 'validated',
+    PENDING: 'pending',
+    REJECTED: 'rejected'
+}
```

Comment:
4 possible state for validating state
1. IDLE => before verification is made
2. VALIDATED => verified
3. PENDING => during verification, when /validate-token is processing
4. REJECTED => not verified